src/apiClient.ts

```diff
@@ -34,34 +34,35 @@ export class TransfaAPIClient {
     method: AxiosRequestConfig["method"],
     endpoint: string,
     data?: any,
-    params?: any,
+    params: Record<string, any> = {},
     headers?: AxiosRequestConfig["headers"]
   ): Promise<AxiosResponse<T>> {
-    if (!params) {
-      params = {};
-    }
 
+    let requestPayload = data
```

Comment:
This line don't prevent the update of the `data` object.

Use the spread operator to clone the object, then verify that the cloning has indeed occurred and that the cloned item is indeed an object.

Reply:
Good point @LewisYann . But data is of type any

Reply:
I will improve the typing later